### PR TITLE
fix: ClusterImages.Validate images

### PR DESCRIPTION
**Pull request by emosbaugh**
## Description

ClusterImages.Validate does not currently call Validate on the nested ImageSpecs resulting in the possibility of including invalid images. This change adds validation for all ImageSpecs under the ClusterImages struct.

Follow up to https://github.com/k0sproject/k0s/pull/4792

Fixes # (issue)

## Type of change

- [x] Bug fix (non-breaking change which fixes an issue)
- [ ] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to not work as expected)
- [ ] Documentation update

## How Has This Been Tested?

- [ ] Manual test
- [x] Auto test added

## Checklist:

- [x] My code follows the style [guidelines](https://github.com/k0sproject/k0s/blob/main/docs/contributors/overview.md) of this project 
- [x] My commit messages are [signed-off](https://github.com/k0sproject/k0s/blob/main/docs/contributors/github_workflow.md)
- [x] I have performed a self-review of my own code
- [x] I have commented my code, particularly in hard-to-understand areas
- [ ] I have made corresponding changes to the documentation
- [x] My changes generate no new warnings
- [x] I have added tests that prove my fix is effective or that my feature works
- [x] New and existing unit tests pass locally with my changes
- [ ] Any dependent changes have been merged and published in downstream modules
- [x] I have checked my code and corrected any misspellings

### CODE REVIEW

File: pkg/apis/k0s/v1beta1/images.go [hunk 1]
```diff
@@ -116,6 +116,17 @@ func (ci *ClusterImages) Validate(path *field.Path) (errs field.ErrorList) {
 		}))
 	}
 
+	errs = append(errs, ci.Konnectivity.Validate(path.Child("konnectivity"))...)
+	errs = append(errs, ci.PushGateway.Validate(path.Child("pushgateway"))...)
+	errs = append(errs, ci.MetricsServer.Validate(path.Child("metricsserver"))...)
+	errs = append(errs, ci.KubeProxy.Validate(path.Child("kubeproxy"))...)
+	errs = append(errs, ci.CoreDNS.Validate(path.Child("coredns"))...)
+	errs = append(errs, ci.Pause.Validate(path.Child("pause"))...)
+	errs = append(errs, ci.Calico.CNI.Validate(path.Child("calico").Child("cni"))...)
```
Comment (twz123) [inline pkg/apis/k0s/v1beta1/images.go:125]: Wondering if it's worth to move that into the CalicoImages / KubeRouterImages structs for consistency.

**Review by twz123 — APPROVED** [review]
Thanks! I've left a few nits that could be fixed if you have the time and inclination.

**Comment by emosbaugh** [comment]
I've addressed the feedback and the change should be ready for another review. Thanks!

**Comment by twz123** [comment]
DCO check is nagging us about 76004b0b148a0216f59fc26cc847f22ab024bf61. Could you please add your --signoff?

**Review by twz123 — APPROVED** [review]
:v:

**Pull request merged**